assembly/src/library/path.rs

```diff
@@ -79,20 +83,27 @@ impl LibraryPath {
     ///
     /// The first component is the leftmost token separated by `::`.
     pub fn first(&self) -> &str {
-        self.path
-            .split_once(Self::PATH_DELIM)
-            .expect("a valid library path must always have at least one component")
-            .0
+        if let Some((first, _)) = self.path.split_once(Self::PATH_DELIM) {
+            first
+        } else {
+            self.path.as_str()
+        }
     }
```

Comment:
Is this change needed to support `mod.masm` files in the root directory of a library? If so, I think we can disallow this.